cocotb/share/lib/gpi_log/gpi_logging.cpp

```diff
@@ -129,18 +129,31 @@ extern "C" void gpi_native_logger_vlog(const char *name, int level,
     log_buff.clear();
     int n = vsnprintf(log_buff.data(), log_buff.capacity(), msg, argp);
     if (n < 0) {
-        // LCOV_EXCL_START
-        fprintf(stderr, "Log message construction failed: (error code) %d\n",
-                n);
-        return;
-        // LCOV_EXCL_STOP
-    } else if ((unsigned)n >= log_buff.capacity()) {
+        // On Windows the above call to vsnprintf will return -1 if the buffer
+        // would overflow, rather than the number of bytes that would be written
+        // (as required by the C99 standard). So we try the call again with the
+        // buffer NULL and the size 0, which should return the number of bytes
+        // that would be written, even on Windows...
+        va_list argp_copy_again;
+        va_copy(argp_copy_again, argp);
```

Comment:
```suggestion
        va_copy(argp_copy_again, argp_copy);
```
`argp` is already used at this point.